storm-core/src/jvm/org/apache/storm/security/auth/kerberos/KerberosSaslTransportPlugin.java

```diff
@@ -50,6 +52,44 @@
 public class KerberosSaslTransportPlugin extends SaslTransportPlugin {
     public static final String KERBEROS = "GSSAPI"; 
     private static final Logger LOG = LoggerFactory.getLogger(KerberosSaslTransportPlugin.class);
+    private static Map <LoginCacheKey, Login> loginCache = new ConcurrentHashMap<>();
```

Comment:
In case people are worried about the cache growing without bound the size of the cache will be proportional to the size of your jaas conf, as the key to the cache is essentially the jaas conf entry for the Login.